### PR TITLE
[WIP] documentation for def fill()

### PR DESCRIPTION
**Pull request by ZohrehPythonSchool**
Hi Nelle,

We have adopted  the documentation of "fill" according to numpy style. I hope you like it!

Best,
Zohreh & Bruno

@NelleV

### CODE REVIEW

**Comment by tacaswell** [comment]
:+1: to the doc fixup, however this put in a mix tabs and spaces in the source.  Can you please replace all of the tabs with 4 spaces?

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -4624,41 +4624,41 @@ def barbs(self, *args, **kw):
...
-        An arbitrary number of *x*, *y*, *color* groups can be specified::
-
-          ax.fill(x1, y1, 'g', x2, y2, 'r')
-
-        Return value is a list of :class:`~matplotlib.patches.Patch`
+        """Plot filled polygons.
+
+	Parameters
+	----------
+        args : a variable length argument, allowing for multiple
```
Comment (tacaswell) [inline lib/matplotlib/axes/_axes.py]: This wrapping is a bit funny.

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -4624,41 +4624,41 @@ def barbs(self, *args, **kw):
...
+            *x*, *y* pairs with an optional color format string; see
+            :func:`~matplotlib.pyplot.plot` for details on the argument
+            parsing.  For example, to plot a polygon with vertices at *x*,
+            *y* in blue.::
+            ax.fill(x,y, 'b' )
```
Comment (tacaswell) [inline lib/matplotlib/axes/_axes.py]: This is missing a blank line and indentation to be picked up as proper rst

File: lib/matplotlib/axes/_axes.py [hunk 3]
```diff
@@ -4624,41 +4624,41 @@ def barbs(self, *args, **kw):
...
 
-        The *closed* kwarg will close the polygon when *True* (default).
-
-        kwargs control the :class:`~matplotlib.patches.Polygon` properties:
-
-        %(Polygon)s
-
-        **Example:**
+       
+        Example
```
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: Here is an extra comma that shouldn't be there.

File: lib/matplotlib/axes/_axes.py [hunk 4]
```diff
@@ -4624,41 +4624,41 @@ def barbs(self, *args, **kw):
...
+            ax.fill(x,y, 'b' )
+            An arbitrary number of *x*, *y*, *color* groups can be specified::
+            ax.fill(x1, y1, 'g', x2, y2, 'r')
+
+ 	kwargs : The *closed* kwarg will close the polygon when *True* (default).
+            kwargs control the :class:`~matplotlib.patches.Polygon` properties:
+            %(Polygon)s
+
+
+        Return
```
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: The correct title for this section is "Returns" with an s

Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: "Return" is [valid in napoleon](http://www.sphinx-doc.org/en/stable/ext/napoleon.html#docstring-sections); haven't found confirmation of NumpyDoc's format yet.

Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: Though it's best to be consistent, of course, if everyone's been writing it one way.

File: lib/matplotlib/axes/_axes.py [hunk 5]
```diff
@@ -4624,41 +4624,41 @@ def barbs(self, *args, **kw):
...
         supports are supported by the fill format string.
 
         If you would like to fill below a curve, e.g., shade a region
         between 0 and *y* along *x*, use :meth:`fill_between`
 
-        The *closed* kwarg will close the polygon when *True* (default).
-
-        kwargs control the :class:`~matplotlib.patches.Polygon` properties:
-
-        %(Polygon)s
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: This has been removed without being replaced.

File: lib/matplotlib/axes/_axes.py [hunk 6]
```diff
@@ -4624,41 +4624,41 @@ def barbs(self, *args, **kw):
...
+        args : a variable length argument, allowing for multiple
+            *x*, *y* pairs with an optional color format string; see
+            :func:`~matplotlib.pyplot.plot` for details on the argument
+            parsing.  For example, to plot a polygon with vertices at *x*,
+            *y* in blue.::
+            ax.fill(x,y, 'b' )
+            An arbitrary number of *x*, *y*, *color* groups can be specified::
+            ax.fill(x1, y1, 'g', x2, y2, 'r')
+
+ 	kwargs : The *closed* kwarg will close the polygon when *True* (default).
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: Keyword arguments should got in a "Keyword Arguments" section, listed individually.

File: lib/matplotlib/axes/_axes.py [hunk 7]
```diff
@@ -4624,41 +4624,41 @@ def barbs(self, *args, **kw):
...
+            kwargs control the :class:`~matplotlib.patches.Polygon` properties:
+            %(Polygon)s
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: `Polygon` will need to be updated to interpolate correctly. It's also indented one too many.

File: lib/matplotlib/axes/_axes.py (hunk 1 above)
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: `args` is not really the name of a parameter; it should be `x` and `y`, but I'm not sure how to indicate that any number of them can be passed.

**Comment by QuLogic** [comment]
It appears you have not [set the config for your git client](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup#Your-Identity). The committer in your commit is attributed to "Your Name <you@example.com>", which is probably not what you want. Note that the _author_ does have correct attribution.

File: lib/matplotlib/axes/_axes.py (hunk 4 above)
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: It is not valid numpydoc.

File: lib/matplotlib/axes/_axes.py (hunk 6 above)
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: There is no keyword arguments in numpydoc, but the kwargs should indeed be moved to a "Other parameters" section.

File: lib/matplotlib/axes/_axes.py (hunk 3 above)
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: Also, the section is called "Examples" (plural). The current name is not rendered properly.

File: lib/matplotlib/axes/_axes.py (hunk 4 above)
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: Hmm, yes, you're right; it'll only be possible when we switch to napoleon.

**Comment by NelleV** [comment]
I did a PR on your branch to fix the small problems left. If you merge the PR, I think we'll be good on this contribution.

**Comment by QuLogic** [comment]
The first commit's attribution includes your first name, but the other two do not. I'm not sure if you want that, but if you are not comfortable with rebasing and changing it, we can perhaps do that for you before merging.

**Comment by ZohrehPythonSchool** [comment]
Hi Nell,

Sorry that it took me so long! I think it should be fine now!

Best,
Zohreh



On 14 Sep 2016, at 04:11, Nelle Varoquaux <notifications@github.com<mailto:notifications@github.com>> wrote:


I did a PR on your branch to fix the small problems left. If you merge the PR, I think we'll be good on this contribution.

**Comment by ZohrehPythonSchool** [comment]
I’m really sorry, I missed this email and did the merge. Is it possible to change my name now? If so, please do feel free to do that.



On 14 Sep 2016, at 04:52, Elliott Sales de Andrade <notifications@github.com<mailto:notifications@github.com>> wrote:


The first commit's attribution includes your first name, but the other two do not. I'm not sure if you want that, but if you are not comfortable with rebasing and changing it, we can perhaps do that for you before merging.

**Comment by QuLogic** [comment]
If @NelleV is fine with the rendering, I can correct the name for you; do you prefer both names or just one?

**Comment by NelleV** [comment]
I've opened a new pull request with the tiny left over fix (#7298). I am thus closing this one.

Thanks for your contribution!

**Pull request closed without merging**